### PR TITLE
components, redux, styling for student and prof

### PR DESCRIPTION
**Pull request by csong202**
- also the `npm run start` wasn't working for me until i changed it
- indentation using tabs (size is 4 spaces) 
- added index.js to make importing components easier
- NewStudentForm, FormGroup, InputGroup, Canvas, ClickableCard, CourseView, SignIn, SignOutBtn, etc components
- added bootstrap
- redux for student (form values for new student registration, list of students) and prof 
- alerts for student sign in page (login success, login failure, new student)
- prof sign in and out
- prof can select a course and search for students taking that course
- single sign in page then the user can choose to sign in as a prof or student
- sign out button in nav bar when user is signed in as a prof

Future Work
- how to display student's attendance record
- multiple profs could teach the same course (need to only search in the course section taught by the current prof). would need more redux for course info
  - or we could ignore that :P 
- redirect to /prof-sign-in when the user tries to navigate to /prof-page while not signed in
- ClickableCard doesn't show very well on the light background. Maybe make it a different colour or use a darker page bg colour (would need to darken box shadow in this case)?

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -6,28 +6,40 @@ const STYLES = ["btn--primary", "btn--outline", "btn--test"];
...
-        onClick={onClick}
-        type={type}
-      >
-        {children}
-      </button>
-    </Link>
-  );
+	const buttonElem = (
+		<button
+			className={`${checkButtonStyle} ${checkButtonSize}${
```
Comment (diegobajetti) [inline src/components/Button.js]: What is the purpose of changing this? Also, the `btn` class is necessary here for the utility classes, all buttons have to be a `btn` class.

File: src/components/FormGroup.css [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (diegobajetti) [inline src/components/FormGroup.css]: File is redundant. Delete it.

File: src/components/HeroSection.js [hunk 3]
```diff
@@ -1,55 +1,57 @@
...
+				autoPlay="autoPlay"
+				loop="loop"
+				muted="muted"
+				playsInline="playsInline"
+			/>
+			<h1>Check My Attendance</h1>
+			<p>with open('check-my-attendance.txt', 'r') as file:</p>
+			<div className="hero-btns">
+				<Button
+					className="btn-in-hero"
```
Comment (diegobajetti) [inline src/components/HeroSection.js]: The class name `btns-in-hero` seems more appropriate.

File: src/components/HeroSection.css [hunk 4]
```diff
@@ -1,104 +1,104 @@
...
 .hero-btns {
-  margin-top: 16px;
+	margin-top: 16px;
 }
 
-.hero-btns .btn {
-  margin: 6px;
-  font-family: "JetBrains Mono", "Trebuchet MS", "Lucida Sans Unicode",
-    "Lucida Grande", "Lucida Sans", Arial, sans-serif;
+.hero-btns .btn-in-hero {
```
Comment (diegobajetti) [inline src/components/HeroSection.css]: The class name `btns-in-hero` seems more appropriate.

File: src/components/Button.js (hunk 1 above)
Comment (csong202) [reply src/components/Button.js]: in case the Button component needs to be used without the link

Comment (csong202) [reply src/components/Button.js]: but then the button component never ended up getting used without the link so I can delete that part

Comment (csong202) [reply src/components/Button.js]: I get this when I add btn
image

Comment (csong202) [reply src/components/Button.js]: image
fixed

File: src/components/HeroSection.css [hunk 5]
```diff
@@ -1,104 +1,104 @@
...
 .hero-btns {
-  margin-top: 16px;
+	margin-top: 16px;
 }
 
-.hero-btns .btn {
-  margin: 6px;
-  font-family: "JetBrains Mono", "Trebuchet MS", "Lucida Sans Unicode",
-    "Lucida Grande", "Lucida Sans", Arial, sans-serif;
+.hero-btns .btns-in-hero {
```
Comment (diegobajetti) [inline src/components/HeroSection.css:40]: Change this back to `btn`. The class specifier here is supposed to refer to the original `btn` utility class in `Button.js`. In this way, any `Button` components used inside the div with classname `hero-btns` have their `btn` property changed.

File: src/components/Navbar.css [hunk 6]
```diff
@@ -1,164 +1,169 @@
...
+
+	.nav-item {
+		height: 80px;
+	}
+
+	.nav-item.button {
+		height: 120px;
+	}
+
+	.nav-nav-menu {
```
Comment (diegobajetti) [inline src/components/Navbar.css]: Redundant name?
```suggestion
	.nav-menu {
```

File: src/components/Navbar.js [hunk 7]
```diff
@@ -1,71 +1,97 @@
...
+							Home
+						</Link>
+					</li>
+					<li className="nav-item">
+						<Link
+							to="/student"
+							className="nav-links"
+							onClick={closeMobileMenu}
+						>
+							Student Sign In
```
Comment (diegobajetti) [inline src/components/Navbar.js]: I think having one button for singing in is better than two individual ones. Maybe once inside the sign in page, the user can choose whether they want to sing in as a prof or user. I can take a look at this.

**Review by diegobajetti — CHANGES_REQUESTED** [review]

File: src/components/Navbar.css (hunk 6 above)
Comment (csong202) [reply src/components/Navbar.css]: i had css issues before i changed the class names

File: src/components/HeroSection.css (hunk 5 above)
Comment (csong202) [reply src/components/HeroSection.css:40]: do we need the buttons in the hero section?

File: src/components/Navbar.js (hunk 7 above)
Comment (csong202) [reply src/components/Navbar.js]: fixed
image

File: src/components/HeroSection.css (hunk 5 above)
Comment (csong202) [reply src/components/HeroSection.css:40]: nvm fixed

File: src/components/Navbar.css (hunk 6 above)
Comment (csong202) [reply src/components/Navbar.css]: nvm fixed

File: src/components/HeroSection.css (hunk 5 above)
Comment (csong202) [reply src/components/HeroSection.css:40]: fixed
image

**Pull request merged**